internal/compose/compose.go

```diff
@@ -1,14 +1,29 @@
 package compose
 
 import (
+	"fmt"
 	"regexp"
+	"strings"
 
 	"github.com/docker/cli/cli/compose/loader"
 	"github.com/docker/cli/cli/compose/template"
 	composetypes "github.com/docker/cli/cli/compose/types"
 	"github.com/pkg/errors"
 )
 
+var (
+	delimiter    = "\\$"
+	substitution = "[_a-z][._a-z0-9]*(?::?[-?][^}]*)?"
```

Comment:
These should be const

Reply:
I agree. It was just a move from render.
But good catch. I will pass `delimiter` and `substitution` to a `const` block.
The other 2 call functions, so I cannot move it